src/Core/Parsers/EdmModelBuilder.cs

```diff
@@ -109,12 +117,19 @@ private EdmModelBuilder BuildEntitySets(ISqlMetadataProvider sqlMetadataProvider
 
             // Entity set is a collection of the same entity, if we think of an entity as a row of data
             // that has a key, then an entity set can be thought of as a table made up of those rows.
-            foreach (KeyValuePair<string, DatabaseObject> entityAndDbObject in sqlMetadataProvider.GetEntityNamesAndDbObjects())
+            IReadOnlyDictionary<string, Entity> linkingEntities = sqlMetadataProvider.GetLinkingEntities();
+            foreach ((string entityName, DatabaseObject dbObject) in sqlMetadataProvider.GetEntityNamesAndDbObjects())
             {
-                if (entityAndDbObject.Value.SourceType != EntitySourceType.StoredProcedure)
+                if (linkingEntities.ContainsKey(entityName))
+                {
+                    // No need to create entity set for linking entity.
```

Comment:
Adding a short note on why entity set need not be created would be helpful.
```suggestion
                    // No need to create entity set for linking entity because the linking entity is not exposed for REST and GraphQL. Therefore, there is no possibility of having a `filter` operation against it.
```